org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java

```diff
@@ -1081,6 +1082,30 @@ public static URI toUri(String uri) {
 		return LSPEclipseUtils.toUri(Path.fromPortableString(URI.create(uri).getPath()));
 	}
 
+	/**
+	 * Use nio Paths to convert a file URI to a File in order to avoid problems with UNC paths on Windows.
+	 * Java has historically generated 'unhealthy' UNC URIs so \\myserver\a\b becomes file:////myserver/a/b
+	 * The favoured representation is to encode the server as the URI authority as file://myserver/a/b
+	 * Java (and LSP4e) has kept the older representation using File.toURI() and new File(URI uri) for
+	 * backward compatibility, but supported the newer representation in nio.Path
+	 * Trying to construct a File directly with a new-style UNC URI causes it to throw with an 'authority is not null'
+	 * complaint. Going via nio.Path allows us to accept either syntax. LSP4e does not use URIs directly e.g. as
+	 * keys in lookup dictionaries so we don't have to worry about canonicalisation problems
+	 *
+	 * See https://bugs.openjdk.org/browse/JDK-4723726
+	 * https://docs.microsoft.com/en-us/archive/blogs/ie/file-uris-in-windows
+	 *
+	 * @param uri A file URI, possibly for a UNC path in the newer syntax with the server encoded in the authority
+	 * @return A file
+	 */
+	public static File fromUri(URI uri) {
```

Comment:
This should be private if it's not used in other classes.

Reply:
Fair point - it's a hang over from when I experimentally tried coaxing LSP4e to work exclusively in terms of the newer syntax (just to see how difficult it would be, ignoring the fact it would probably break some java-based lang servers). We already had a few conversion methods in this utils class so I wanted it to be a public One True Way to convert URIs to Files. If we only want to be tolerant of the newer syntax, then you're right there is no consumer outside this class itself.

Reply:
What do you think about testing? Unfortunately the java APIs themselves are platform dependent so you can't just even write a pure unit test that will work cross platform. I can use JUnit Assume.that() but it won't be much use if there is no build infrastructure that ever runs on Windows.

Reply:
It's up to you. If you prefer adding a test, that would be welcome. And it's very likely that someday someone contributes an automated test execution on windows (it's not that hard), so those tests may be helpful in the future.
But if you prefer, we can merge without those tests; it will help already also things would be more likely to see regressions in the future.

Reply:
I added a unit test but that meant making fromURI public again as package-private isn't enough for test accessibility due to the code structure...

I looked into an integration test but that's going to be very difficult given the way we're set up - can't realistically set up a network share mid test, can't use the local admin share if the workspace isn't mounted in that way in the first place, and can't use mocking very easily because of the prevalence of static methods in the Eclipse APIs used :-(